https://chromium-review.googlesource.com/c/chromium/src/+/4926575
From: Benjamin Beaudry <benjamin.beaudry@microsoft.com>
Date: Tue, 10 Oct 2023 20:10:22 +0000
Subject: [PATCH] [a11y] Fix GetClosestElement to skip over pseudo element descendants

(cherry picked from commit 1136ca6c870d1d1c05bd70353ca5c347acc38b41)

Fixed: 1486913
Change-Id: I622dacb69be310a032e7ebd92c7f680525606a11
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4894915
Reviewed-by: Aaron Leventhal <aleventhal@chromium.org>
Reviewed-by: Nektarios Paisios <nektar@chromium.org>
Commit-Queue: Benjamin Beaudry <benjamin.beaudry@microsoft.com>
Cr-Original-Commit-Position: refs/heads/main@{#1202882}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4926575
Cr-Commit-Position: refs/branch-heads/5993@{#1238}
Cr-Branched-From: 511350718e646be62331ae9d7213d10ec320d514-refs/heads/main@{#1192594}
--- a/third_party/blink/renderer/modules/accessibility/ax_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_object.cc
@@ -5620,8 +5620,12 @@
   if (!element) {
     for (AXObject* parent = ParentObject(); parent;
          parent = parent->ParentObject()) {
-      if (parent) {
-        return parent->GetElement();
+      // It's possible to have a parent without a node here if the parent is a
+      // pseudo element descendant. Since we're looking for the nearest element,
+      // keep going up the ancestor chain until we find a parent that has one.
+      element = parent->GetElement();
+      if (element) {
+        return element;
       }
     }
   }
